java/src/main/java/ai/rapids/cudf/JCudfSerialization.java

```diff
@@ -1037,7 +1037,11 @@ private static long copySlicedOffsets(DataWriter out, ColumnBufferProvider colum
       // Don't copy anything, there are no rows
       return 0;
     }
-    long bytesToCopy = (numRows + 1) * 4;
+    long bytesToCopy = 0;
+    if (column.getHostBufferFor(ColumnVector.BufferType.DATA) != null) {
+      bytesToCopy = (numRows + 1) * 4;
```

Comment:
This is adding in the number of bytes needed to copy the offsets.  That will never change for a string.  We are only looking at the data portion, which was already not being copied.

Reply:
I see. Thanks for pointing out. I thought my test needed this due to a failure to get array length later in sliceAndPad but I verified that my test was not in any need for this change. Removed.